mobly/controllers/android_device_lib/snippet_client.py

```diff
@@ -65,21 +67,36 @@ def _do_start_app(self):
 
     def stop_app(self):
         """Overrides superclass."""
-        cmd = _STOP_CMD.format(self.package)
+        cmd = _STOP_CMD % self.package
         logging.info('Stopping snippet apk with: %s', cmd)
-        out = self._adb.shell(_STOP_CMD.format(self.package)).decode('utf-8')
+        out = self._adb.shell(_STOP_CMD % self.package).decode('utf-8')
         if 'OK (0 tests)' not in out:
             raise Error('Failed to stop existing apk. Unexpected output: %s' %
                         out)
 
-    def _is_app_installed(self):
+    def check_app_installed(self):
```

Comment:
jsonrpc_client_base has unit tests. Since you're expanding the installation checking logic in this PR, this is where the unit test should be added too. :)